pymodbus/client/base.py

```diff
@@ -191,6 +191,14 @@ async def async_execute(self, request=None):
         packet = self.framer.buildPacket(request)
         self.transport_send(packet)
         if self.params.broadcast_enable and not request.slave_id:
+            if self.comm_params.handle_local_echo:
```

Comment:
This will not work in all cases. Local echo is something that might happen, but are not guaranteed to happen in all cases.

You mix transport and client, which secures your PR will not work for the server.

Reply:
If the local echo doesn't happen then the future will be completed anyway and this will work.

There is a test in the suite for when a local echo does not occur.

Reply:
Yes but the future should go away, echo is handled solely in the transport class! Read the definition of the transport class.